packages/core/parcel-bundler/src/transforms/babel/jsx.js

```diff
@@ -37,7 +57,11 @@ async function getJSXConfig(asset, isSourceModule) {
     }
   }
 
-  if (pragma || JSX_EXTENSIONS[path.extname(asset.name)]) {
+  if (
+    pragma ||
+    JSX_EXTENSIONS[path.extname(asset.name)] ||
+    isUsingJSXinJS(asset)
```

Comment:
Should we also use the regex to determine what pragma to use? Presumably this case is to detect if you haven't already installed one of those dependencies yet.

Reply:
This ia a valid concern, I was thinking about it as well. Luckily, the pipeline is built in a way that it would, after creating ast in pretransform, re-collect dependencies, install them and re-iterate over that function which results in setting correct pragma.

On second thought, I've decided to implement your suggestion as it should add robustness.